syntax = "proto3";

message Spaceship {
  enum Alignment {
    Ally = 0;
    Enemy = 1;
  }
  enum Class {
    Corvette = 0;
    Frigate = 1;
    Cruiser = 2;
    Destroyer = 3;
    Carrier = 4;
    Dreadnought = 5;
  }
  message Officer {
    string first_name = 1;
    string last_name = 2;
    string rank = 3;
  }
  optional Alignment alignment = 1;
  optional string name = 2;
  optional Class class_ = 4;
  optional float length = 3;
  optional int32 crew_size = 5;
  optional bool armed = 6;
  repeated Officer officers = 7;
}

message Coordinates { repeated float coordinate = 1; }

service SpaceshipService {
  rpc GetSpaceships(Coordinates) returns (stream Spaceship) {}
}
